internal/controller/instaslice_controller.go

```diff
@@ -264,6 +264,12 @@ func (r *InstasliceReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 			}
 			log.Info("finalizer deleted for succeeded ", "pod", pod.Name)
 		}
+		// If no allocations exist, update metrics with all slots free
+		err = r.updateMetricsAllSlotsFree(ctx, instasliceList)
```

Comment:
I think `updateMetricsAllSlotsFree` is called for cases when pods succeeded, can you please confirm what will happen when pods fail,ie logic starting in line 188